File: backend/internal/tests/endpoints.go
package tests

type Endpoint struct {
	ID       string
	Name     string
	Path     string
	Category string
}

var frontendEndpoints = []Endpoint{
	{ID: "root", Name: "Root Page", Path: "/", Category: "performance"},
	{ID: "large", Name: "Large File", Path: "/large-probe.txt", Category: "performance"},
	{ID: "small", Name: "Small File", Path: "/probe.txt", Category: "performance"},
	{ID: "cache-time", Name: "Cache Headers", Path: "/api/time", Category: "caching"},
	{ID: "cache-bypass", Name: "Cache Bypass", Path: "/cache/bypass/nocache", Category: "caching"},
	{ID: "sql", Name: "Security - SQL", Path: "/security/sql/union", Category: "security"},
	{ID: "xss", Name: "Security - XSS", Path: "/security/xss/script", Category: "security"},
	{ID: "redirect", Name: "Redirect 301", Path: "/redirect/301", Category: "features"},
}

var apiEndpoints = []Endpoint{
	{ID: "api-domains", Name: "API: List Domains", Path: "/api-test/domains", Category: "api"},
	{ID: "api-domain-details", Name: "API: Domain Details", Path: "/api-test/domain-details", Category: "api"},
	{ID: "api-ssl", Name: "API: SSL Settings", Path: "/api-test/ssl", Category: "api"},
	{ID: "api-dns", Name: "API: DNS Records", Path: "/api-test/dns", Category: "api"},
	{ID: "api-caching", Name: "API: Cache Settings", Path: "/api-test/caching", Category: "api"},
	{ID: "api-firewall", Name: "API: Firewall Rules", Path: "/api-test/firewall", Category: "api"},
	{ID: "api-analytics", Name: "API: Traffic Reports", Path: "/api-test/analytics", Category: "api"},
}

func AllEndpoints() []Endpoint {
	return append(append([]Endpoint{}, frontendEndpoints...), apiEndpoints...)
}

func IsAPICategory(category string) bool {
	return category == "api"
}

func FrontendEndpoints() []Endpoint {
	return append([]Endpoint{}, frontendEndpoints...)
}

func APIEndpoints() []Endpoint {
	return append([]Endpoint{}, apiEndpoints...)
}

File: frontend/src/data/endpoints.js
const frontendEndpoints = [
  { id: 'root', name: 'Root Page', path: '/', category: 'performance' },
  { id: 'large', name: 'Large File', path: '/large-probe.txt', category: 'performance' },
  { id: 'small', name: 'Small File', path: '/probe.txt', category: 'performance' },
  { id: 'cache-time', name: 'Cache Headers', path: '/api/time', category: 'caching' },
  { id: 'cache-bypass', name: 'Cache Bypass', path: '/cache/bypass/nocache', category: 'caching' },
  { id: 'sql', name: 'Security - SQL', path: '/security/sql/union', category: 'security' },
  { id: 'xss', name: 'Security - XSS', path: '/security/xss/script', category: 'security' },
  { id: 'redirect', name: 'Redirect 301', path: '/redirect/301', category: 'features' }
];

const apiEndpoints = [
  { id: 'api-domains', name: 'API: List Domains', path: '/api-test/domains', category: 'api' },
  { id: 'api-domain-details', name: 'API: Domain Details', path: '/api-test/domain-details', category: 'api' },
  { id: 'api-ssl', name: 'API: SSL Settings', path: '/api-test/ssl', category: 'api' },
  { id: 'api-dns', name: 'API: DNS Records', path: '/api-test/dns', category: 'api' },
  { id: 'api-caching', name: 'API: Cache Settings', path: '/api-test/caching', category: 'api' },
  { id: 'api-firewall', name: 'API: Firewall Rules', path: '/api-test/firewall', category: 'api' },
  { id: 'api-analytics', name: 'API: Traffic Reports', path: '/api-test/analytics', category: 'api' }
];

export function getAllEndpoints() {
  return [...frontendEndpoints, ...apiEndpoints];
}

export function getCategoryDisplayName(category) {
  const names = {
    performance: '⚡ Frontend Performance',
    caching: '💾 Caching & Headers',
    security: '🛡️ Security & WAF',
    features: '🔧 Additional Features',
    api: '🔌 API Functionality'
  };
  return names[category] || category;
}

